### PR TITLE
50141_initialize_file_uploader

### PR DESCRIPTION
**Pull request by manazirahsan**

### CODE REVIEW

**Comment by abdullahallmehedi** [comment]
ok

**Comment by SSalekin** [comment]
ok

File: app/models/book.rb [hunk 1]
```diff
@@ -1,7 +1,8 @@
 class Book < ActiveRecord::Base
   belongs_to :category
-  has_many :book_images, dependent: :destroy
   has_many :users_books, dependent: :destroy
-  has_many :reviews, dependent: :destroy
   has_many :users, through: :users_books
+  has_many :book_images, dependent: :destroy
+  has_many :reviews, dependent: :destroy
+  accepts_nested_attributes_for :book_images
```
Comment (dieunb) [inline app/models/book.rb]: should split it to blocks code with relation. It is define for nested attributes, so split it to new block. And add condition to reject it.

File: app/models/book_image.rb [hunk 2]
```diff
@@ -1,3 +1,4 @@
 class BookImage < ActiveRecord::Base
   belongs_to :book
+  mount_uploader :image, ImageUploader
```
Comment (dieunb) [inline app/models/book_image.rb]: same above

File: app/uploaders/image_uploader.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  # Process files as they are uploaded:
+  # process :scale => [200, 300]
+  #
+  # def scale(width, height)
+  #   # do something
+  # end
+
+  # Create different versions of your uploaded files:
+  version :thumb do
+    process :resize_to_fit => [50, 50]
```
Comment (dieunb) [inline app/uploaders/image_uploader.rb]: Convention

File: bin/setup [hunk 4]
```diff
@@ -14,7 +14,7 @@ Dir.chdir APP_ROOT do
 
   # puts "\n== Copying sample files =="
   # unless File.exist?("config/database.yml")
-  #   system "cp config/database.yml.sample config/database.yml"
```
Comment (dieunb) [inline bin/setup]: checkout this file

File: app/models/book.rb (hunk 1 above)
Comment (manazirahsan) [inline app/models/book.rb]: well, i will add condition to reject it.
but what is a "new block"?

**Comment by manazirahsan** [comment]
allow_destroy and reject_if added to BookImage.
bin/setup is checkedOut.

**Comment by abdullahallmehedi** [comment]
ok

**Comment by SSalekin** [comment]
ok

**Pull request merged**